app/uk/gov/hmrc/gform/sharedmodel/DataRetrieveDefinition.scala

```diff
@@ -337,6 +337,46 @@ object DataRetrieveDefinitions {
     Map(DataRetrieve.Attribute("riskScore") -> DataRetrieve.AttrType.Integer)
   )
 
+  val hmrcTaxRates = DataRetrieveDefinition(
+    DataRetrieve.Type("hmrcTaxRates"),
+    Attr.FromObject(
+      List(
+        AttributeInstruction(
+          DataRetrieve.Attribute("regime"),
+          ConstructAttribute.AsIs(Fetch(List("regime")))
+        ),
+        AttributeInstruction(
+          DataRetrieve.Attribute("code"),
+          ConstructAttribute.AsIs(Fetch(List("code")))
+        ),
+        AttributeInstruction(
+          DataRetrieve.Attribute("rate"),
+          ConstructAttribute.AsIs(Fetch(List("rate")))
+        ),
+        AttributeInstruction(
+          DataRetrieve.Attribute("startDate"),
+          ConstructAttribute.AsIs(Fetch(List("startDate")))
+        ),
+        AttributeInstruction(
+          DataRetrieve.Attribute("endDate"),
+          ConstructAttribute.AsIs(Fetch(List("endDate")))
+        )
+      )
+    ),
+    List(
+      Parameter("regime"),
+      Parameter("code"),
+      Parameter("date")
+    ),
+    Map(
+      DataRetrieve.Attribute("regime")    -> DataRetrieve.AttrType.String,
+      DataRetrieve.Attribute("code")      -> DataRetrieve.AttrType.String,
+      DataRetrieve.Attribute("rate")      -> DataRetrieve.AttrType.String,
```

Comment:
This can't be `String`, otherwise expression like `dataRetrieve.todaysRate.rate` would evaluate to a `StringResult` and `${dataRetrieve.todaysRate.rate * 1.2}` would not work

Reply:
I've now added a BigDecimal type

Reply:
Ah okay - I've just tested this with the changes and it certainly works now.